ui/apps/platform/src/Containers/Vulnerabilities/VulnerablityReporting/api/useCreateReport.ts

```diff
@@ -36,87 +31,10 @@ function useCreateReport(): CreateReportResult {
-
-        // transform form values to values to be sent through API
-        const fixability: Fixability =
-            reportParameters.cveStatus.length > 1 ? 'BOTH' : reportParameters.cveStatus[0];
-
-        const vulnReportFiltersBase: VulnerabilityReportFiltersBase = {
-            fixability,
-            severities: reportParameters.cveSeverities,
-            imageTypes: reportParameters.imageType,
-        };
-        let vulnReportFilters: VulnerabilityReportFilters;
-        if (reportParameters.cvesDiscoveredSince === 'SINCE_LAST_REPORT') {
-            vulnReportFilters = {
-                ...vulnReportFiltersBase,
-                lastSuccessfulReport: true,
-            };
-        } else if (
-            reportParameters.cvesDiscoveredSince === 'START_DATE' &&
-            reportParameters.cvesDiscoveredStartDate
-        ) {
-            vulnReportFilters = {
-                ...vulnReportFiltersBase,
-                startDate: new Date(reportParameters.cvesDiscoveredStartDate).toISOString(),
-            };
-        } else {
-            vulnReportFilters = {
-                ...vulnReportFiltersBase,
-                allVuln: true,
-            };
-        }
-
-        const notifiers = deliveryDestinations.map((deliveryDestination) => {
-            return {
-                emailConfig: {
-                    notifierId: deliveryDestination.notifier?.id || '',
-                    mailingLists: deliveryDestination.mailingLists,
-                },
-                notifierName: '',
-            };
-        });
-
-        let schedule: Schedule;
-        if (formSchedule.intervalType === 'WEEKLY') {
-            schedule = {
-                intervalType: 'WEEKLY',
-                hour: 0,
-                minute: 0,
-                daysOfWeek: {
-                    days: formSchedule.daysOfWeek.map((day) => Number(day)),
-                },
-            };
-        } else {
-            schedule = {
-                intervalType: 'MONTHLY',
-                hour: 0,
-                minute: 0,
-                daysOfMonth: {
-                    days: formSchedule.daysOfMonth.map((day) => Number(day)),
-                },
-            };
-        }
-
-        const reportData: ReportConfiguration = {
-            id: '',
-            name: reportParameters.reportName,
-            description: reportParameters.description,
-            type: 'VULNERABILITY',
-            vulnReportFilters,
-            resourceScope: {
-                collectionScope: {
-                    collectionId: reportParameters.reportScope?.id || '',
-                    collectionName: reportParameters.reportScope?.name || '',
-                },
-            },
-            notifiers,
-            schedule,
-        };
+        const reportConfiguration = getReportConfigurationFromFormValues(formValues);
 
         // send API call
-        createReportConfiguration(reportData)
+        createReportConfiguration(reportConfiguration)
```

Comment:
Related nit to a comment above: any time I have a function where the last thing it does is execute a promise, I tend to return the promise even if the result isn't currently used and there is code to handle the `then` and `catch` callbacks like there is here. I think it gives more flexibility to the calling function to have the ability to wait on the async call if needed.